source/kraken/worker.cpp

```
@@ -919,11 +919,16 @@ void Worker::street_network_routing_matrix(const pbnavitia::StreetNetworkRouting
 
         auto* row = this->pb_creator.mutable_sn_routing_matrix()->add_rows();
         for(auto coord : dest_coords) {
+            auto* k = row->add_routing_response();
             auto it = nearest.find(coord.uri());
-            if (it != nearest.end()) {
-                row->add_duration(it->second.total_seconds());
-            }else {
-                row->add_duration(-1);
+            assert(it != nearest.end());
```

why can you assert this ?

---

maybe throw a recoverable exception so we can see it in logs in release

---

OK